### DISCUSSION TITLE
Spatio-temporal GAM vs sdmTMB Model Standard Errors

### DISCUSSION THREAD

**Discussion by Nathan-Hebert** [sdmTMB/sdmTMB#179] [Q&A] [answered]
I'm fitting a binomial spatio-temporal SDM in both mgcv and sdmTMB. One model is of course using the SPDE approach, and the other is using spatio-temporal smooth functions. I notice that the GMRF standard errors (on the link scale) change less throughout time when compared to their GAM counterparts. I'm just curious if anybody has any idea why. Is this difference due to GMRFs vs splines?

(The color scale on each is the same)
![Rplot01](https://user-images.githubusercontent.com/62037477/219074980-964a86e3-6652-4461-9cca-9bd79d9f6aba.png)
![Rplot](https://user-images.githubusercontent.com/62037477/219074987-0f28658b-88ad-433e-ae5f-081ea7df6bd3.png)

**Comment by ericward-noaa** [comment]
It'd be interesting to see the structure of each of the models you're using because there might be subtle differences between the GAM and SPDE implementation (e.g. did you include spatial + spatiotemporal effects? just spatial?) 

There certainly look to be differences (why can't the GAM predict to the dark grey areas for example?). One reason may be the distribution of spline knots relative to the data is not as uniform as the SPDE mesh? 

If you haven't seen it, Dave Miller has an awesome paper diving into this comparison more (and it includes code!): https://link.springer.com/article/10.1007/s13253-019-00377-z

**Comment by Nathan-Hebert** [comment]
Sorry about the confusion on the dark grey areas! ggplot2 cut that bit off because its off the color scale I used for the other image! :)

I used spatio-temporal fields in sdmTMB and a function of (eastings, northings, year) in the GAM. I'm intrigued by your comment regarding the distribution of the spline knots. I wonder if there's a way to check that...

**Comment by ericward-noaa** [comment]
Yes -- it should be pretty easy to extract. How you do it depends on the type of spline you're using

```
x <- rnorm(30)
y <- -0.1*x + 0.01*x*x
```

If you're using p-splines or b-splines
```
fit <- gam(y ~ s(x, bs = 'ps', k = 20)) # can also be 'bs'
knot_locs <- fit$smooth[[1]]$knots
```

For a cubic regression spline:
```
fit <- gam(y ~ s(x, bs = 'cr, k = 20)) 
knot_locs <- fit$smooth[[1]]$xp 
```

Gavin Simpson has a few good blog posts around knots -- and this example shows how to specify knot locations to the `knots` argument https://fromthebottomoftheheap.net/2020/06/03/extrapolating-with-gams/

**Comment by seananderson** [comment]
For the GAM, it could depend on whether you have `s(X, Y, by = year_factor)` or `s(X, Y, by = year)` or `s(X, Y, year)` (or, preferably, the `te()` equivalent). Assuming you have the latter 3-dimensional smooth, I would expect constructing the smoother over time to add more variability in uncertainty over time compared to having repeated slices each year. It wouldn't be uncommon to have more uncertainty at the beginning and end of the time smoother.

**Comment by Nathan-Hebert** [comment]
Yeah, its the te() version of s(X, Y, year). TPS, which I think makes things tricky because it doesn't use knots in the traditional way as far as I understand.

I suppose using year in an sdmTMB model with spatio-temporal fields would essentially be a repeated slice each year.

**Comment by seananderson** [comment, marked as answer]
If 2001 and 2019 are the beginning and end (or near to it) then might you just be seeing the uncertainty from the smoother tails?

``` r
library(mgcv)
#> Loading required package: nlme
#> This is mgcv 1.8-41. For overview type 'help("mgcv-package")'.
fit <- gam(density ~ s(year, k = 3), data = sdmTMB::pcod, family = tw())
plot(fit)
```

![](https://i.imgur.com/QC86ppr.png)

<sup>Created on 2023-02-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by Nathan-Hebert** [comment]
Interesting. Yeah, they are the beginning and end years.
